diff --git a/lib/compiler/rspack-compiler.js b/lib/compiler/rspack-compiler.js
index d6b778456240df013903cbf9f49c33064d980503..c5e1901e0f01965870186622661a18f59a6c1fa2 100644
--- a/lib/compiler/rspack-compiler.js
+++ b/lib/compiler/rspack-compiler.js
@@ -1,6 +1,6 @@
 import { existsSync } from 'fs';
 import { createRequire } from 'module';
-import { join } from 'path';
+import { join, resolve } from 'path';
 import { ERROR_PREFIX, INFO_PREFIX } from '../ui/index.js';
 import { isEsmProject } from '../utils/is-esm-project.js';
 import { BaseCompiler } from './base-compiler.js';
@@ -13,7 +13,7 @@ export class RspackCompiler extends BaseCompiler {
     }
     run(configuration, tsConfigPath, appName, extras, onSuccess) {
         const cwd = process.cwd();
-        const configPath = join(cwd, tsConfigPath);
+        const configPath = resolve(cwd, tsConfigPath);
         if (!existsSync(configPath)) {
             throw new Error(`Could not find TypeScript configuration file "${tsConfigPath}".`);
         }
@@ -21,7 +21,7 @@ export class RspackCompiler extends BaseCompiler {
         const pathToSource = this.getPathToSource(configuration, tsConfigPath, appName);
         const entryFile = getValueOrDefault(configuration, 'entryFile', appName, 'entryFile', extras.options);
         const entryFileRoot = getValueOrDefault(configuration, 'root', appName) || '';
-        const defaultOptions = rspackDefaultsFactory(pathToSource, entryFileRoot, entryFile, extras.debug ?? false, tsConfigPath, plugins, isEsmProject(), extras.tsOptions);
+        const defaultOptions = rspackDefaultsFactory(pathToSource, entryFileRoot, entryFile, extras.debug ?? false, configPath, plugins, isEsmProject(), extras.tsOptions);
         let rspack;
         try {
             rspack = require('@rspack/core');
